magento/magento2
# Fix AbstractFieldArray with column renderers, Call setId() and setName(), remove workarounds

### Description (*)
See https://github.com/magento/magento2/issues/22008.

The name and ID attributes aren't correctly set on custom renderers, and then implementations have all kinds of workarounds for that and require complex custom overrides to set the default attribute.

### Fixed Issues (if relevant)

1. Fixes magento/magento2#22008: Columns in AbstractFieldArray subclasses do not get their ID set correctly
2. ...

### Manual testing scenarios (*)

1. Go to Stores > Configuration  > Catalog > Inventory
2. Ensure that the Minimum Qty Allowed in Shopping Cart is working in the same way with all the removed code in \Magento\CatalogInventory\Block\Adminhtml\Form\Field\Minsaleqty

### Contribution checklist (*)
